V0.27  typo tolerance configuration (#1606)
meilisearch/documentation: learn/configuration/settings.md

| **[sortableAttributes](/learn/configuration/settings.md#sortable-attributes)**    | [Strings] | Attributes to use when [sorting](/learn/advanced/sorting.md) search results  | `[]`                                                                         |
| **[stopWords](/learn/configuration/settings.md#stop-words)**                       | List of words ignored by Meilisearch when present in search queries              | `[]`                                                                                        |
| **[synonyms](/learn/configuration/settings.md#synonyms)**                          | List of associated words treated similarly                                       | `{}`                                                                                        |

> | **[typoTolerance](/learn/configuration/settings.md#typo-tolerance)**               |Typo tolerance settings                                             | `{}`                                                                                        |

Reviewer: It's tricky as well because most of the other settings are strings or arrays, but this one is an object. So we can't really use the same format as the other short descriptions in the table.

I tried to think of a descriptive phrase and came up with `Adjust conditions for typo tolerance or disable it on certain words or attributes`

However, I'm honestly wondering if it would be wiser to go with something generic. I can see this feature evolving quite a bit with new forms of customization. So something like this could be better for longevity: `Object used to customize typo tolerance behavior`.

<CodeSamples id="settings_guide_synonyms_1" />

By doing so, when searching for `black sweater`, results for `black jumper` will also be returned.

## Typo tolerance

The typo tolerance settings for an index. Typo tolerance allows users to make mistakes when typing and still find something. [Disabling typo tolerance will not impact the `typo` ranking rule.](/learn/configuration/typo_tolerance.md#impact-of-typo-tolerance-on-the-typo-ranking-rule)

> The `typoTolerance` object is composed of the following fields:

Author: I'm not sure if this is the best way to summarize this, but any suggestions are welcome

Reviewer 2: I would have slightly changed the first sentence.

"The typo tolerance settings for an index. Typo tolerance allows users to make mistakes when typing and still find documents containing words close to their typed query"

- `enabled`
- `disableOnAttributes`
- `disableOnWords`
- `minWordSizeForTypos`

[Learn more about typo tolerance](/learn/configuration/typo_tolerance.md)

### Example

> By adding the following settings, you will disable typo tolerance for `title` and set the minimum words size for 2 typos to 10 characters.

Author: Does this example need to show how to use each field?

<CodeSamples id="settings_guide_synonyms_1" />

By doing so, when searching for `black sweater`, results for `black jumper` will also be returned.

## Typo tolerance

> The typo tolerance settings for an index. Typo tolerance allows users to make mistakes when typing and still find results close to their query.

Reviewer: I like this suggestion from a marketing standpoint but prefer Maryam's text for a documentation context. I think my reaction stems from "ensures" and "always"; these words strike me as promises that we can't keep and shouldn't make (in docs—in another context this would be great copy).

Made my own little Frankenstein from both of your texts. Thanks!

```suggestion
The typo tolerance settings for an index. Typo tolerance helps users find relevant results even when their queries contain spelling mistakes or typos.
```

> The `typoTolerance` object allows you to configure the following:

Reviewer: ```suggestion
The `typoTolerance` object allows you to:
```

> - enable or disable the typo tolerance feature with the `enabled` field
> - set the minimum word size for typos  with `minWordSizeForTypos`
> - disable typos on specific words with `disableOnWords`
> - disable typos on specific document attributes with `disableOnAttributes`

Reviewer: ```suggestion
- enable or disable the typo tolerance feature with the `enabled` field
- configure the minimum word size for typos to be handled with `minWordSizeForTypos`
- disable typos on specific words with `disableOnWords`
- disable typos on specific document attributes with `disableOnAttributes`
```

> [Disabling typo tolerance will not impact the `typo` ranking rule.](/learn/configuration/typo_tolerance.md#impact-of-typo-tolerance-on-the-typo-ranking-rule)

Reviewer: Great question and one not answered by [the spec](https://github.com/meilisearch/specifications/pull/117/files#diff-af7982d588ce2d959088ed71f2b17ed36d9cb4eb1add31aaeb2c32194c4355baR32).

My guess is that if `typoTolerance` is disabled, the `typo` rule does nothing. 

Reasoning: with `typoTolerance` disabled, only exact prefix matches will be found. These will all contain 0 typos and thus be equal under the `typo` rule, so it effectively does nothing and just passes to the next ranking rule in the bucket sort.

@gmourier Can you confirm?

> [Disabling typo tolerance will not impact the `typo` ranking rule.](/learn/configuration/typo_tolerance.md#impact-of-typo-tolerance-on-the-typo-ranking-rule)

Reviewer: Related comment: https://github.com/meilisearch/documentation/pull/1606#discussion_r862957031

> [Disabling typo tolerance will not impact the `typo` ranking rule.](/learn/configuration/typo_tolerance.md#impact-of-typo-tolerance-on-the-typo-ranking-rule)

Reviewer: Confirmed via [private Slack thread](https://meilisearch.slack.com/archives/C01BDAZ8GFQ/p1651740061567789?thread_ts=1651738360.942419&cid=C01BDAZ8GFQ).

[Learn more about typo tolerance](/learn/configuration/typo_tolerance.md)

### Example

> By adding the following settings, you will disable typo tolerance for `title` and set the minimum words size for 2 typos to `10` characters.

Reviewer: ```suggestion
Adding the following settings disables typo tolerance for the `title` attribute and sets the minimum word size for 2 typos to `12` characters.
```
